### PR TITLE
use BinaryBuilder for binaries

### PR DESCRIPTION
**Pull request by KristofferC**
Not sure if this is wanted but it was a fun exercise to try out the new stuff.

I can put back the source build fallback if people want.

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -1,38 +1,50 @@
...
+    if haskey(download_info, platform_key())
+        url, tarball_hash = download_info[platform_key()]
+        if unsatisfied || !isinstalled(url, tarball_hash; prefix=prefix)
+            # Download and install binaries
+            install(url, tarball_hash; prefix=prefix, force=true, verbose=verbose)
         end
+    elseif unsatisfied
+        # If we don't have a BinaryProvider-compatible .tar.gz to download, complain.
+        # Alternatively, you could attempt to install from a separate provider,
+        # build from source or something even more ambitious here.
```
Comment (tkelman) [inline deps/build.jl]: a source build fallback is needed, or this drops support for source builds of julia on distros that use gcc < 7. that includes the fedora packages, and likely juliapro as well.

**Comment by andreasnoack** [comment]
Looks like there is an issue with the source build on nightly

**Comment by KristofferC** [comment]
I get the following locally:

```
ld: unknown option: -z
collect2: error: ld returned 1 exit status
make: *** [libopenspecfun.1.3.dylib] Error 1
ERROR: failed process: Process(`make install ARCH=x86_64 'CC=gcc -m64' 'FC=gfortran -m64' USECLANG=0 USEGCC=1 USE_OPENLIBM=1 'CFLAGS=-O3 -std=c99' 'FFLAGS=-O2 -fPIC' "LDFLAGS=-L/Users/kristoffer/.julia/v0.6/SpecialFunctions/deps/usr/lib -Wl,-rpath,'\$\$ORIGIN' -Wl,-z,origin" DESTDIR= prefix= libdir=/Users/kristoffer/.julia/v0.6/SpecialFunctions/deps/usr/lib includedir=/Users/kristoffer/.julia/v0.6/SpecialFunctions/deps/usr/include O=`, ProcessExited(2)) [2]
```

The fact that the build log file doesn't show in CI is really annoying...  Gotta update Base Pkg with https://github.com/JuliaLang/Pkg.jl/pull/380...

**Comment by KristofferC** [comment]
The source build passes on Travis linux now. Some problems with installing gfortran on Travis mac but not sure it will help since the error in the post above might appear on the mac builder as well (I have a mac locally).

File: .travis.yml [hunk 2]
```diff
@@ -2,9 +2,20 @@ language: julia
 os:
   - linux
   - osx
+addons:
+  apt_packages:
+    - gfortran
+before_install:
+  - if [ "$TRAVIS_OS_NAME" == "osx" ]; then
+        brew install gcc;
```
Comment (andreasnoack) [inline .travis.yml]: Isn't `gcc` already installed?

**Pull request merged**